### PR TITLE
METRON-1847 break out functions from prepare-commit for re-use

### PR DESCRIPTION
**Pull request by ottobackwards**
This pr breaks out the prepare-commit script into a common script and a prepare-commit script.  This will allow re-use of that code for future scripts.

Test by preparing..... commits for current PR's as is your fashion.  The script should work as before.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [-] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [-] Have you written or updated unit tests and or integration tests to verify your changes?
- [-] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [-] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [-] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: dev-utilities/committer-utils/prepare-commit [hunk 1]
```diff
@@ -16,242 +16,31 @@
...
 
-# working directory
-WORK=~/tmp/${CHOSEN_REPO}-pr$PR
-read -p "  local working directory [$WORK]: " INPUT
-[ -n "$INPUT" ] && WORK=$INPUT
+run_tests
 
-# handle tilde expansion
-WORK="${WORK/#\~/$HOME}"
+please_review_commit_then
```
Comment (JonZeolla) [inline dev-utilities/committer-utils/prepare-commit]: s/commit_//

File: dev-utilities/committer-utils/metron-committer-common [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+      # write setting to config file, so it is not needed next time
+      echo "APACHE_EMAIL=$APACHE_EMAIL" >> ${CONFIG_FILE}
+    fi
+}
+
+#
+# Provide the user with a choice of the metron or bro repositories
+#
+function chose_metron_or_bro_repo {
```
Comment (JonZeolla) [inline dev-utilities/committer-utils/metron-committer-common]: choose?

File: dev-utilities/committer-utils/prepare-commit [hunk 3]
```diff
@@ -16,242 +16,31 @@
...
 #
 
-# not likely to change
-METRON_UPSTREAM="https://git-wip-us.apache.org/repos/asf/metron.git"
-BRO_PLUGIN_UPSTREAM="https://git-wip-us.apache.org/repos/asf/metron-bro-plugin-kafka.git"
-CONFIG_FILE=~/.metron-prepare-commit
-GITHUB_REMOTE="origin"
-BASE_BRANCH=master
+LOCATION_DIR=`dirname $0`
+#shellcheck source=./metron-committer-common
```
Comment (nickwallen) [inline dev-utilities/committer-utils/prepare-commit]: Does this need to be commented out?

Comment (ottobackwards) [reply dev-utilities/committer-utils/prepare-commit]: Yes, this a directive for shellcheck

https://github.com/koalaman/shellcheck/wiki/SC1090
https://github.com/koalaman/shellcheck/wiki/Directive

File: dev-utilities/committer-utils/prepare-commit [hunk 4]
```diff
@@ -16,242 +16,31 @@
 #  limitations under the License.
 #
 
-# not likely to change
-METRON_UPSTREAM="https://git-wip-us.apache.org/repos/asf/metron.git"
-BRO_PLUGIN_UPSTREAM="https://git-wip-us.apache.org/repos/asf/metron-bro-plugin-kafka.git"
-CONFIG_FILE=~/.metron-prepare-commit
-GITHUB_REMOTE="origin"
-BASE_BRANCH=master
+LOCATION_DIR=`dirname $0`
```
[the remark this answers is not in the document]
Comment (ottobackwards) [reply dev-utilities/committer-utils/prepare-commit]: Why cd there?  how about
LOCATION_DIR="$( dirname "${BASH_SOURCE[0]}"

[the remark this answers is not in the document]
Comment (ottobackwards) [reply dev-utilities/committer-utils/prepare-commit]: nm, you are right

**Pull request closed without merging**